src/rust.rs

```diff
@@ -495,6 +495,22 @@ impl HandleValue {
     }
 }
 
+impl HandleValueArray {
+    pub fn new() -> HandleValueArray {
+        HandleValueArray {
+            length_: 0,
+            elements_: ptr::null_mut(),
+        }
+    }
+
+    pub fn from_slice(values: &[Value]) -> HandleValueArray {
```

Comment:
This is not clearly safe to me - it's easy to create a vector which isn't rooted, leading to a handle that can point to things that get GCed. We should either remove this constructor, figure out a way to make it safe, or mark it unsafe.